src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -181,6 +183,29 @@ function MoneyRequestReportTransactionList({
         [reportActions, sortedTransactions],
     );
 
+    useEffect(() => {
+        const lastFullScreenRoutes = navigationRef.getRootState()?.routes.findLast((route) => isFullScreenName(route.name));
+        if (lastFullScreenRoutes?.name !== NAVIGATORS.REPORTS_SPLIT_NAVIGATOR && lastFullScreenRoutes?.name !== NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR) {
+            return;
+        }
+
+        const lastRoute = lastFullScreenRoutes?.state?.routes?.at(-1);
+        if (!lastRoute?.params) {
+            return;
+        }
```

Comment:
Could we do this instead?
```suggestion
        if (!lastRoute?.params || !('reportID' in lastRoute.params)) {
            return;
        }
```

Then we can shorten this condition here:
```javascript
        const reportIDParams = 'reportID' in lastRoute.params ? lastRoute.params.reportID : '';
        if (reportIDParams !== report.reportID) {
            return;
        }
```
to:

```javascript
        if (lastRoute.params.reportID !== report.reportID) {
            return;
        }

```